src/FromJSONTrait.php

```diff
@@ -27,9 +27,10 @@ public static function fromJSON($jsonStr) {
 
         if (is_null($cfg)) {
             $err = json_last_error();
-            throw new \InvalidArgumentException("Invalid JSON: $err");
+            throw new JSONParseErrorException($cfg, $err, json_last_error_msg());
```

Comment:
Based on the signature of the above exception shouldn't `$cfg` be `$jsonStr`? I think it should be the valued we tried to JSON decode, $cfg is the value as decoded (which based on the if check will always be `null`).